src/sinks/datadog/logs/sink.rs

```diff
@@ -368,8 +363,14 @@ where
                 let fut = svc.call(req)
                     .err_into()
                     .map(move |result| {
-                        logic.update_finalizers(result, finalizers);
-
+                        let status = match result {
+                            Err(error) => {
+                                error!("Sink IO failed with error: {}", error);
+                                EventStatus::Failed
+                            },
+                            Ok(response) => { response.into() }
+                        };
+                        finalizers.update_status(status);
```

Comment:
More of a thought: I wonder if we'll ever want a layer where the response is passed on to something else, necessitating getting the event status of a response by ref rather than consuming it entirely.

Reply:
I feel pretty confident that we will. Absent something concrete I wasn't quite sure how I wanted to build to handle that. Honestly, I was hoping the s3 work I allude to would force that.